lib/firehose/server/publisher.rb

```diff
@@ -132,8 +132,13 @@ def eval_publish_script(channel_key, message, ttl, buffer_size, persist, deferra
         local message_payload = firehose_resource .. payload_delimiter .. sequence .. payload_delimiter .. message
 
         redis.call('set', sequence_key, sequence)
-        redis.call('lpush', list_key, message)
-        redis.call('ltrim', list_key, 0, buffer_size - 1)
+        if buffer_size > 0 then
+          redis.call('lpush', list_key, message)
+          redis.call('ltrim', list_key, 0, buffer_size - 1)
+        else
+          redis.call('del', list_key')
```

Comment:
It's failing on this line, there's an extra ' at the end before the closing paren (syntax error when being parsed for execution in redis).

Reply:
Ah good catch! Thanks!